### DISCUSSION TITLE
Upgrade library from v1.4.1 to v1.9.x

### DISCUSSION THREAD

**Discussion by boxedcode** [kube-hetzner/terraform-hcloud-kube-hetzner#553] [Q&A] [answered]
Hi there,

I'm currently using version `v1.4.1` of your library, and I would like to upgrade it to version `v1.9.x`. My question is, are the releases backwards compatible, or will the upgrade require me to effectively create a new cluster?

I apologise if the library upgrade process is documented somewhere and I've missed it.

Thanks,

**Comment by aleksasiriski** [comment, marked as answer]
You can safely upgrade, I would recommend looking at the latest `kube.tf.example` which may include new options that could not be wanted enabled by default.

**Reply by mysticaltech** [reply]
Exactly as @aleksasiriski said. Just change the version to the latest one, and adapt the new kube.tf to match your config, and apply. You should not see too many errors!

If you do see errors, like in the LB name for instance, that changed since then, just git clone the latest version, and in `source`, point to it instead of the terraform cloud module name.

Then, you can easily go to the required file and you can make temporary changes to match your current config so that terraform applies, then you just point it back to the module and apply again. That should fix everything!

Now, there is another option, which is to deploy a new cluster, and migrate your old one with Velero! Just search the project for this for more info.

**Reply by mysticaltech** [reply]
You can also run `terraform destroy -target 'module.kube-hetzner.null_resource.kustomization'` before changing the version and adapting the kube.tf file. This will ease everything out, as barely anything has changed outside of that.

And no, it's not destructive on the cluster, it will just enable you to re-apply the changes to higher-level components to match the new kube.tf when you later upgrade the version (more on this in the readme).

**Reply by mysticaltech** [reply]
Also, have a look at https://github.com/orgs/kube-hetzner/discussions/497
